packages/fiori-elements-writer/src/data/manifestSettings.ts

```diff
@@ -0,0 +1,54 @@
+import { FioriElementsApp, TemplateType } from '../types';
+import { OdataVersion } from '@sap-ux/odata-service-writer';
+import { Editor } from 'mem-fs-editor';
+import { join } from 'path';
+import { render } from 'ejs';
+
+/**
+ * Extend the manifest.json file passed via the fs reference with the template and settings specified.
+ * This extends by combining the version specific common manifest with the floorplan/template type
+ * specific settings.
+ *
+ * @param fs - A reference to the in memory file system
+ * @param targetPath - The target application path
+ * @param rootTemplatesPath - The root path where the manifest templates will be found
+ * @param feApp - The applicaiton config to be used by the manifest templates
+ */
+export function extendManifestJson<T>(
+    fs: Editor,
+    targetPath: string,
+    rootTemplatesPath: string,
+    feApp: FioriElementsApp<T>
+) {
+    let templatePath = feApp.template.type;
+    // FEOP and ALP v4 are variants of LROP and so we use the same template
+    if (
+        feApp.service?.version === OdataVersion.v4 &&
+        [TemplateType.FormEntryObjectPage, TemplateType.AnalyticalListPage, TemplateType.ListReportObjectPage].includes(
+            feApp.template.type
+        )
+    ) {
+        templatePath = TemplateType.ListReportObjectPage;
+    }
+
+    // Enhance template settings
+    const templateSettings = {
+        ...feApp.template.settings,
+        filterModel: feApp.service?.model,
+        type: feApp.template.type
+    };
+
+    // Manifest paths to be extended
+    const extendTemplatePaths = [
+        join(rootTemplatesPath, `v${feApp.service?.version}`, templatePath, 'extend', 'webapp'),
+        join(rootTemplatesPath, `v${feApp.service?.version}`, 'common', 'extend', 'webapp')
```

Comment:
What happens when `feApp.service` is `undefined`?

Reply:
It's a mandatory property, it should not be undefined. We do not currently validate inputs thoroughly. Full validation needs discussion with @tobiasqueck. If the service is not defined you should not be able to create an FE app.